### PR TITLE
[C++ API Parity] Add xor_convergence test for lbfgs

### PR DESCRIPTION
**Pull request by anjali411**

### CODE REVIEW

File: test/cpp/api/optim.cpp [hunk 1]
```diff
@@ -198,6 +206,10 @@ TEST(OptimTest, XORConvergence_SGD) {
       SGDOptions(0.1).momentum(0.9).nesterov(true).weight_decay(1e-6)));
 }
 
+TEST(OptimTest, XORConvergence_LBFGS) {
+  ASSERT_TRUE(test_optimizer_xor<LBFGS>(LBFGSOptions(1.0).line_search_fn("strong_wolfe")));
```
Comment (yf225) [inline test/cpp/api/optim.cpp]: Would `ASSERT_TRUE(test_optimizer_xor<LBFGS>(LBFGSOptions(1.0)));` also work? XD

Comment (anjali411) [reply test/cpp/api/optim.cpp]: great idea! added the test for optim without line_search_fn :D

**Review by yf225 — APPROVED** [review]
Thanks for the awesome work!

**Comment by mrshenli** [comment]
Looks like this PR breaks master @anjali411 

https://app.circleci.com/pipelines/github/pytorch/pytorch/142016/workflows/58f4e55c-7806-4db8-b48e-1d87a3ccc08e/jobs/4864445

**Comment by yf225** [comment]
I tried `const int64_t kBatchSize = 50;` and it seems to work locally (not sure why `kBatchSize = 6` was failing locally for me)

**Pull request closed without merging**